specs/experimental/fault-proof/stage-one/honest-challenger-fdg.md

```diff
@@ -33,145 +33,72 @@ Dispute Games.
-depth in the range of `[1, MAX_DEPTH]`, the honest challenger processes them and performs
-a response.
-
-To determine the appropriate response, the challenger first needs to know which
-[_team_](fault-dispute-game.md#team-dynamics) it belongs to.
-This determines the set of claims it should respond to in the FDG.
-If the agent determines itself to be a Defender, which aims to support the root claim,
-then it must dispute claims positioned at odd depths in the game tree.
-Otherwise, it disputes claims positioned at even depths in the game tree.
-This means an honest challenger will typically only respond to claims made by the opposing team.
-(See [Countering Freeloaders](#countering-freeloaders) for exceptions to this).
-
-The next step is to determine if the claim, now known to be for the opposing team,
-disputes another claim the honest challenger _agrees_ with.
-An honest challenger agrees with a claim iff every other claim along its path to the
-root claim commits to a valid `ClaimHash`. Put differently, an honest challenger will
-avoid countering a claim if it disagrees with the path of claims leading to that
-specific claim. But if the honest challenger agrees with the path leading to the claim,
-then the claim is countered.
-
-The last step is to determine whether the claim has a valid commitment (i.e. `ClaimHash`).
-If the `ClaimHash` matches the honest challenger's at the same trace index, then we
-disagree with the claim's stance by moving to [defend](fault-dispute-game.md#defend).
+of the game being interacted with to generate the traces needed to make truthful claims.
+
+## Invariants
+
+To ensure an accurate and incentive compatible fault dispute system, the honest challenger behaviour must preserve
+three invariants for any game:
+
+1. The game resolves as `DefenderWins` if the root claim is correct and `ChallengerWins` if the root claim is incorrect
+2. The honest challenger is refunded the bond for every claim it posts and paid the bond of the parent of that claim
+3. The honest challenger never counters its own claim
+
+## Fault Dispute Game Responses
+
+The honest challenger determines which claims to counter by iterating through the claims in the order they are stored
+in the contract. This ordering ensures that a claim's ancestors are processed prior to the claim itself. For each claim,
+the honest challenger determines and tracks what the honest response would be, regardless of whether that response
+already exists in the full game state.
+
+The root claim is considered to be an honest claim if and only if it has a [ClaimHash](fault-dispute-game.md#claims)
+that agrees with the honest challenger's ClaimHash for the root claim.
+
+The honest challenger should counter a claim if and only if:
+
+1. The claim is a child of an honest claim
+2. The claim has an honest sibling with a trace index greater than or equal to the claim's trace index
```

Comment:
Let's clarify that the "honest sibling" here doesn't necessarily mean an actual sibling in the game (which may not yet exist), but an honest sibling that should exist after processing the claim's parent.

Reply:
Good call out, I've reworded them both to be based on the set of honest responses. The first rule works either way but seems worth being consistent.